pennylane/estimator/ops/op_math/symbolic.py

```diff
@@ -151,14 +151,16 @@ def resource_decomp(cls, base_cmpr_op: CompressedResourceOp, **kwargs):
 
         """
         base_class, base_params = (base_cmpr_op.op_type, base_cmpr_op.params)
-        base_params = {key: value for key, value in base_params.items() if value is not None}
-        kwargs = {key: value for key, value in kwargs.items() if key not in base_params}
+
+        for key, value in kwargs.items():
+            if key in base_params and base_params[key] is None:
+                base_params[key] = value
```

Comment:
```suggestion
        base_params.update(
    (k, v) for k, v in kwargs.items() if k in base_params and base_params[k] is None
)
```

Will need to format this